packages/pigeon/lib/java_generator.dart

```diff
@@ -162,21 +176,25 @@ static MessageCodec<Object> getCodec() {
           indent.scoped('{', '} else {', () {
             indent.write('channel.setMessageHandler((message, reply) -> ');
             indent.scoped('{', '});', () {
-              final String argType = method.argType;
-              final String returnType = method.returnType;
+              final String argType = _boxedType(method.argType);
+              final String returnType = _boxedType(method.returnType);
               indent.writeln('Map<String, Object> wrapped = new HashMap<>();');
               indent.write('try ');
               indent.scoped('{', '}', () {
                 final List<String> methodArgument = <String>[];
                 if (argType != 'void') {
                   indent.writeln('@SuppressWarnings("ConstantConditions")');
-                  indent.writeln(
-                      '$argType input = $argType.fromMap((Map<String, Object>)message);');
+                  indent.writeln('$argType input = ($argType)message;');
```

Comment:
I think I'm missing some high-level context here; why isn't the map lookup still required for other types?

Reply:
Last week we moved to the codegen custom codec.  The codec will do the converting to/from a map for objects.  That opened the door to allow any argument that the codec supports to be sent.